src/store/contracts/actions.ts

```diff
@@ -1,9 +1,12 @@
 import { createAsyncThunk, createAction } from '@reduxjs/toolkit';
 import { Contract } from '@models/contract.interface';
 import { fetchContract as fetchContractFromApi } from '@common/api/contracts';
+import { selectCurrentNetworkUrl } from '@store/ui/selectors';
 
-export const fetchContract = createAsyncThunk<Contract, string>('contracts/fetch', async query =>
-  fetchContractFromApi(query)
+export const fetchContract = createAsyncThunk<Contract, string>(
+  'contracts/fetch',
+  // @ts-ignore
+  async (query, { getState }) => fetchContractFromApi(selectCurrentNetworkUrl(getState()))(query)
```

Comment:
```suggestion
  async (query, { getState }) =>
    fetchContractFromApi(selectCurrentNetworkUrl(getState() as any) as string)(query)
```

This casting is preferable to ts-ignore, since even thought it bypasses this particular type issue, it doesn't forgot all type safty on this line.